File: 2843. Count Symmetric Integers.cpp
class Solution {
public:
    int countSymmetricIntegers(int low, int high) {
        int ans=0;
        for(int i=low;i<=high;i++){
            string str=to_string(i);
            if(str.length()%2==1)   continue;
            int half=str.length()/2;
            int leftsum=0,rightsum=0;
            for(int i=0;i<half;i++){
                leftsum+=str[i]-'0';
            }
            for(int i=half;i<str.length();i++){
                rightsum+=str[i]-'0';
            }
            if(leftsum==rightsum)   ans++;
        }
        return ans;
    }
};

File: 2843. Count Symmetric Integers.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        ans=0
        for i in range(low,high+1):
            s=str(i)
            n=len(s)
            if n%2 == 1:
                continue
            half=n//2
            leftsum=sum(int(s[i]) for i in range(half))
            rightsum=sum(int(s[i]) for i in range(half,n))
            if leftsum == rightsum:
                ans+=1
        return ans